src/lib/update-lock.ts

```diff
@@ -92,54 +65,32 @@ export const readLock: LockFn = Bluebird.promisify(locker.async.readLock, {
 });
 
 function dispose(release: () => void): Bluebird<void> {
-	return Bluebird.map(_.keys(locksTaken), (lockName) => {
-		delete locksTaken[lockName];
-		return lockFile.unlockAsync(lockName);
+	return Bluebird.map(lockfile.getLocksTaken(), (lockName) => {
+		return lockfile.unlock(lockName);
 	})
 		.finally(release)
 		.return();
 }
 
-const lockExistsErrHandler = (err: Error, release: () => void) => {
-	let errMsg = err.message;
-	if (EEXIST(err)) {
-		// Extract appId|appUuid and serviceName from lockfile path for log message
-		// appId: [0-9]{7}, appUuid: [0-9a-w]{32}, short appUuid: [0-9a-w]{7}
-		const pathMatch = err.message.match(
-			/\/([0-9]{7}|[0-9a-w]{32}|[0-9a-w]{7})\/(.*)\/(?:resin-)?updates.lock/,
-		);
-		if (pathMatch && pathMatch.length === 3) {
-			errMsg = `Lockfile exists for ${JSON.stringify({
-				serviceName: pathMatch[2],
-				[/^[0-9]{7}$/.test(pathMatch[1]) ? 'appId' : 'appUuid']: pathMatch[1],
-			})}`;
-		}
-	}
-	return dispose(release).throw(new UpdatesLockedError(errMsg));
-};
-
 /**
  * Try to take the locks for an application. If force is set, it will remove
  * all existing lockfiles before performing the operation
  *
- * TODO: convert to native Promises. May require native implementation of Bluebird's dispose / using
+ * TODO: convert to native Promises and async/await. May require native implementation of Bluebird's dispose / using
  *
  * TODO: Remove skipLock as it's not a good interface. If lock is called it should try to take the lock
  * without an option to skip.
  */
 export function lock<T extends unknown>(
-	appId: number | null,
+	appId: number,
 	{ force = false, skipLock = false }: { force: boolean; skipLock?: boolean },
 	fn: () => Resolvable<T>,
 ): Bluebird<T> {
-	if (skipLock) {
+	if (skipLock || appId == null) {
 		return Bluebird.resolve(fn());
 	}
 
 	const takeTheLock = () => {
-		if (appId == null) {
```

Comment:
Since you removed this, the `if` in line 252 is no longer necessary right?

Reply:
I'm not sure which file you're referring to?

Reply:
I meant the `if` in

```
if (disposer) {
  return Bluebird.using(disposer, fn as () => PromiseLike<T>);
} else {
  return Bluebird.resolve(fn());
}
```

Since `takeTheLock` never returns undefined, then you don't need this branching.